site/content/blog/2024-10-23-zap-contributor-license-agreement/index.md

```diff
@@ -0,0 +1,60 @@
+---
+title: "ZAP Contributor License Agreement"
+summary: >
+  We are introducing a Contributor License Agreement to cover all ZAP contributions.
+type: post
+tags:
+- blog
+- hacking
+date: "2024-01-23"
+authors: 
+    - simon
+---
+
+We are in the process of introducing a ZAP [Contributor License Agreement](http://oss-watch.ac.uk/resources/cla),
+which is something many other large open source projects have in place.
+
+For now we are only introducing an "Individual CLA" - we are happy to add a "Corporate CLA" if the need arises.
+
+### Why Add a CLA?
+
+As stated in the link above:
+
+> The purpose of a CLA is to ensure that the guardian of a project’s outputs has the necessary ownership or grants of rights over all contributions to allow them to distribute under the chosen licence. In some cases this will mean that the contributor will assign the copyright in all contributions to the project owner; in other cases, they will grant an irrevocable licence to allow the project maintainer to use the contribution. CLAs also have roles in raising awareness of IPR issues within a project.
+    
+### The CLA Pull Request
+
+The CLA will be introduced by PR [#8317](https://github.com/zaproxy/zaproxy/pull/8317). The actual wording of the CLA is heavily based on the [Netty Project Individual CLA](https://docs.google.com/forms/d/e/1FAIpQLSd7Bzje39G__THDJLRgKpQZ4gODNE26x_hZW3ofQOkgL6RGCA/viewform?formkey=dHBjc1YzdWhsZERUQnhlSklsbG1KT1E6MQ).
+
+This PR adds a new action which runs when you submit a PR to the [zaproxy](https://github.com/zaproxy/zaproxy) repo.
+One or more PRs will then add the action to the other ZAP repos. 
+
+### The CLA Process
+
+We're trying to make this as lightweight as possible.
+
+The new "CLA Assistant" action uses the [cla-assistant](https://github.com/cla-assistant/cla-assistant) action and checks to 
+see if the author of the PR has already signed the CLA. If they have not then it will fail the PR, explaining why:
+
+![CLA Screenshot](images/cla.png)
+
+As you will see, all you will need to do is to reply stating: 
+
+    I have read the CLA Document and I hereby sign the CLA
+    
+That it!
+
+Hopefully you will agree that is pretty lightweight :wink:
+
+The fact you have agreed to the CLA is recorded in a private repo - the only personal details recorded is your GitHub username.
+
+### Past Contributions
+
+If you have contributed to ZAP in the past we would appreciate it if you would sign the CLA.
+You can do that by submitting a dummy PR to any of the ZAP repos and then replying to the prompt as above.
+We don't need the PR to be merged so can then just close it.
```

Comment:
Shall we create a repo for this specifically? `cla-signing` or something like that, just to keep the history/noise confined.

Reply:
Might be better. We could make the [cla](https://github.com/zaproxy/cla) repo public and use that I guess?
Theres nothing sensitive in there, it will just need a quick tidy up...

Reply:
Sure.

Reply:
Updated.
We will need to merge https://github.com/zaproxy/cla/pull/2 and make that repo public before publishing this blog post..

Reply:
Now https://github.com/zaproxy/cla/pull/3 😉